Mathlib/AlgebraicGeometry/EllipticCurve/Group.lean

```diff
@@ -95,27 +95,143 @@ abbrev FunctionField : Type u :=
 
 namespace CoordinateRing
 
-section Ring
+section Algebra
 
-/-! ### Ideals in the coordinate ring over a ring -/
+/-! ### The coordinate ring as an `R[X]`-algebra -/
 
-instance instIsDomainCoordinateRing [IsDomain R] [NormalizedGCDMonoid R] :
-    IsDomain W.CoordinateRing :=
-  (Quotient.isDomain_iff_prime _).mpr <| by
-    simpa only [span_singleton_prime W.polynomial_ne_zero, ← irreducible_iff_prime]
-      using W.irreducible_polynomial
-#align weierstrass_curve.coordinate_ring.is_domain WeierstrassCurve.Affine.CoordinateRing.instIsDomainCoordinateRing
+noncomputable instance : Algebra R W.CoordinateRing :=
+  Quotient.algebra R
+#align weierstrass_curve.coordinate_ring.algebra' WeierstrassCurve.Affine.CoordinateRing.instAlgebra
 
-instance instIsDomainCoordinateRing_of_Field {F : Type u} [Field F] (W : Affine F) :
-    IsDomain W.CoordinateRing := by
-  classical exact instIsDomainCoordinateRing W
-#align weierstrass_curve.coordinate_ring.is_domain_of_field WeierstrassCurve.Affine.CoordinateRing.instIsDomainCoordinateRing_of_Field
+noncomputable instance : Algebra R[X] W.CoordinateRing :=
+  Quotient.algebra R[X]
+#align weierstrass_curve.coordinate_ring.algebra WeierstrassCurve.Affine.CoordinateRing.instAlgebraPolynomial
+
+instance : IsScalarTower R R[X] W.CoordinateRing :=
+  Quotient.isScalarTower R R[X] _
+#align weierstrass_curve.coordinate_ring.is_scalar_tower WeierstrassCurve.Affine.CoordinateRing.instIsScalarTowerPolynomial
+
+instance [Subsingleton R] : Subsingleton W.CoordinateRing :=
```

Comment:
According to the naming convention, this should be `WeierstrassCurve.Affine.CoordinateRing.insSubsingleton`, so
```suggestion
instance instSubsingleton [Subsingleton R] : Subsingleton W.CoordinateRing :=
```

Reply:
I confirm that this is the auto-generated name. The full list of names are:
WeierstrassCurve.Affine.CoordinateRing.instAlgebra
WeierstrassCurve.Affine.CoordinateRing.instAlgebraPolynomial
WeierstrassCurve.Affine.CoordinateRing.instIsScalarTowerPolynomial
WeierstrassCurve.Affine.CoordinateRing.instSubsingleton
WeierstrassCurve.Affine.CoordinateRing.instIsDomain

Reply:
I don't think there is a style convention that we should manually name instances? Now that the instance name generation actually produces sensible names, I do feel that we should let it do its job whenever possible.

Reply:
Oh sorry, I thought the autogenerated name now was something like `Blah.instSubsingleton_mathlib`